appservice/src/intTest/groovy/nu/yona/server/CreateUserOnBuddyRequestTest.groovy

```diff
@@ -112,6 +114,122 @@ class CreateUserOnBuddyRequestTest extends AbstractAppServiceIntegrationTest
 		appService.deleteUser(updatedBob)
 	}
 
+	def 'Bob adjusts data, including his Android device'()
+	{
+		given:
+		def richard = addRichard()
+		def mobileNumberBob = makeMobileNumber(timestamp)
+		def inviteUrl = buildInviteUrl(sendBuddyRequestForBob(richard, mobileNumberBob))
+		def bob = appService.getUser(appService.&assertUserGetResponseDetailsWithPrivateDataCreatedOnBuddyRequest, inviteUrl, true, null)
+
+		when:
+		def newNickname = "Bobby"
+		def updatedBobJson = bob.convertToJson()
```

Comment:
Maybe a `def bobToBeUpdated = bob.clone()` would be easier to read:
```
def bobToBeUpdated = bob.clone()
bobToBeUpdated.nickname = newNickname
bobToBeUpdated.deviceName = "My S8"
...
```

Reply:
That won't work as the nickname is a final field. I'd rather leave it that way. I favor immutability over the simplicity of cloning.

Reply:
OK